sql/catalyst/src/main/scala/org/apache/spark/sql/types/AbstractDataType.scala

```diff
@@ -114,6 +114,15 @@ private[sql] object TypeCollection {
     BooleanType,
     ByteType, ShortType, IntegerType, LongType)
 
+  /**
+   * Types that include numeric types and interval type. They are only used in unary_minus,
+   * unary_positive, add and subtract operations.
+   */
+  val NumericAndInterval = TypeCollection(
+    ByteType, ShortType, IntegerType, LongType,
+    FloatType, DoubleType, DecimalType,
+    IntervalType)
```

Comment:
OK. Another problem that might be minor is that the casting error message will become `argument 1 is expected to be of type (numeric or interval)`, instead of `...type (tinyint or smallint or int or bigint or float or double or decimal or interval...`.

Is it still informative?